wcfsetup/install/files/lib/system/message/embedded/object/MessageEmbeddedObjectManager.class.php

```diff
@@ -283,10 +289,36 @@ public function getContentLanguageID() {
 	 * @param	integer		$languageID
 	 */
 	public function setActiveMessage($messageObjectType, $messageID, $languageID = null) {
+		if ($this->activeMessageObjectTypeID) {
+			$this->acitveMessageHistory[] = [
+				'activeMessageID' => $this->activeMessageID,
+				'activeMessageLanguageID' => $this->activeMessageLanguageID,
+				'activeMessageObjectTypeID' => $this->activeMessageObjectTypeID,
+			];
+		}
+
 		$this->activeMessageObjectTypeID = ObjectTypeCache::getInstance()->getObjectTypeIDByName('com.woltlab.wcf.message', $messageObjectType);
 		$this->activeMessageID = $messageID;
 		$this->activeMessageLanguageID = $languageID;
 	}
+
+	/**
+	 * Restores the internal state in case of nested message processing.
+	 */
+	public function reset() {
+		$newState = array_pop($this->acitveMessageHistory);
```

Comment:
When merging this up: Don't forget to add the `\` (or simply also add it for 5.3).